src/db/sysdb_subdomains.c

```diff
@@ -218,6 +225,13 @@ check_subdom_config_file(struct confdb_ctx *confdb,
           sd_conf_path, CONFDB_DOMAIN_FQ,
           subdomain->fqnames ? "TRUE" : "FALSE");
 
+    /* use fully qualified names as output in order to avoid causing
+     * conflicts with users who have the same name and either the
+     * shortname user resolution is enabled or the trusted domain has
+     * been explicitly set to use non-fully qualified names as input.
+     */
+    subdomain->output_fqnames = true;
+
```

Comment:
Do we actually need this? This code is only ever called from new_subdomain which already sets output_fqnames to false and there is no code (except for creating a new domain) which would set the option to true.